portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/BullhornServiceImpl.java

```diff
@@ -251,58 +276,83 @@ public void update(Observable o, final Object arg) {
+                                            countCache.remove(to);
+                                        }
                                     }
                                 }
-                            }
-                        } catch (IdUnusedException idue) {
+                            } catch (IdUnusedException idue) {
                             log.error("No site with id '" + siteId + "'", idue);
                         } finally {
                             lock(sa);
                         }
-                    } else if (AssignmentConstants.EVENT_ADD_ASSIGNMENT.equals(event)) {
-                        String siteId = pathParts[3];
-                        String assignmentId = pathParts[pathParts.length - 1];
-                        SecurityAdvisor sa = unlock(new String[] {AssignmentServiceConstants.SECURE_ACCESS_ASSIGNMENT, AssignmentServiceConstants.SECURE_ADD_ASSIGNMENT_SUBMISSION});
-                        try {
-                            Assignment assignment = assignmentService.getAssignment(assignmentId);
-                            Instant openTime = assignment.getOpenDate();
-                            if (openTime == null || openTime.isBefore(Instant.now())) {
-                                Site site = siteService.getSite(siteId);
-                                String title = assignment.getTitle();
-                                String url = assignmentService.getDeepLink(siteId, assignmentId);
-                                // Get all the members of the site with read ability
-                                for (String  to : site.getUsersIsAllowed(AssignmentServiceConstants.SECURE_ACCESS_ASSIGNMENT)) {
-                                    if (!from.equals(to) && !securityService.isSuperUser(to)) {
-                                        doAcademicInsert(from, to, event, ref, title, siteId, e.getEventTime(), url);
-                                        countCache.remove(to);
+                    } else if (AssignmentConstants.EVENT_ADD_ASSIGNMENT.equals(event) || AssignmentConstants.EVENT_POST_ASSIGNMENT.equals(event)) {
+
+                            String siteId = pathParts[3];
+                            String assignmentId = pathParts[pathParts.length - 1];
+                            SecurityAdvisor sa = unlock(new String[] {AssignmentServiceConstants.SECURE_ACCESS_ASSIGNMENT, AssignmentServiceConstants.SECURE_ADD_ASSIGNMENT_SUBMISSION});
+                            switchToAdmin();
+                            try {
+                                Assignment assignment = assignmentService.getAssignment(assignmentId);
+                                switchToNull();
+                                Instant openTime = assignment.getOpenDate();
+                                if (openTime == null || openTime.isBefore(Instant.now())) {
+                                    Site site = siteService.getSite(siteId);
```

Comment:
`siteService.getSite()` can throw an exception. This should probably be wrapped in a `try/catch`.